test/integration/kubernetes.test.ts

```diff
@@ -226,6 +241,45 @@ tap.test('snyk-monitor pulls images from a private ECR and sends data to kuberne
     'snyk-monitor sent expected data to upstream in the expected timeframe');
 });
 
+tap.test('snyk-monitor pulls images from a local registry and sends data to kubernetes-upstream', async (t) => {
+  t.plan(3);
+
+  const deploymentName = 'python-local';
+  const namespace = 'services';
+  const clusterName = 'Default cluster';
+  const deploymentType = WorkloadKind.Deployment;
+  const imageName = 'kind-registry:5000/python:rc-buster';
+
+  await kubectl.waitForJob('push-to-registry', 'services');
+  
+  console.log('Applying local registry workload...');
+  await kubectl.applyK8sYaml('./test/fixtures/insecure-registries/python-local-deployment.yaml');
+
+  console.log(`Begin polling upstream for the expected kind-registry:5000 image with integration ID ${integrationId}...`);
+
+  const validatorFn: WorkloadLocatorValidator = (workloads) => {
+    return workloads !== undefined &&
+      workloads.find((workload) => workload.name === deploymentName &&
+        workload.type === WorkloadKind.Deployment) !== undefined;
+  };
+
+  const testResult = await validateUpstreamStoredData(
+    validatorFn, `api/v2/workloads/${integrationId}/${clusterName}/${namespace}`);
+  t.ok(testResult, 'snyk-monitor sent expected data to upstream in the expected timeframe');
+
+  const depGraphResult = await getUpstreamResponseBody(
+    `api/v1/dependency-graphs/${integrationId}/${clusterName}/${namespace}/${deploymentType}/${deploymentName}`);
+  
+  t.ok('dependencyGraphResults' in depGraphResult,
+  'expected dependencyGraphResults field to exist in /dependency-graphs response');
+
+  /* Because of a bug in removeTagFromImage() func in src/scanner/images/index.ts, 
+  which chops off everything after ':' from the image name, we store a wrong image name 
+  and the result does not exist in the object referred below */
+  t.same(depGraphResult.dependencyGraphResults[imageName], null,
+  'expected result for image kind-registry:5000/python:rc-buster does not exist');
```

Comment:
Nice! Could we also show how it chops it off? So we prove the bug exists and when we fix it, we can remove this case? Otherwise the comparison to null could just mean that we didn't send the data!
```js
t.ok('kind-registry' in depGraphResult.dependencyGraphResults, 'BUG: the full image name is not stored in kubernetes-upstream');
```